scripts/get_liquidity_params.py

```diff
@@ -1,170 +1,200 @@
-                              P_0_tokens_to_eth=token_options["p0"],
-                              num_formula_precision_bits=40,
-                              max_cap_buy_eth=3.0,
-                              max_cap_sell_eth=3.0,
-                              fee_in_precents=0.25)
-
-
-        # calculate ptoential minimal and maximal ratios according to fix amounts the rm is willing to put (100 eth, 1M tokens).
-        (minimal_pmin_ratio, maximal_pmax_ratio) = liq.calc_pmin_pmax_ratio_based_on_balances(E_0_ether=100.0, T_0_tokens=1000000)
-        # actually use pmin and pmax as fixed 0.5 and 2.0
-        (pmin_ratio, pmax_ratio) = (0.5, 2.0)
-        liq.calc_params_based_on_ratios(pmin_ratio, pmax_ratio)
-
-        #E_0_ether, T_0_tokens = liq.calc_balances_based_on_pmin_pmax(
-        #    min_token_to_eth_rate=option["min_token_to_eth_ratio"] * token_options["p0"],
-        #    max_token_to_eth_rate=option["max_token_to_eth_ratio"] * token_options["p0"]
-        #)
-
-        #liq.calc_params_based_on_ratios(option["min_token_to_eth_ratio"],
-        #                                option["max_token_to_eth_ratio"])
+from pprint import pprint
+
+
+def calc_balances_from_pmin_pmax(
+    rate,
+    price,
+    min_ratio,
+    max_ratio
+):
+
+    min_rate = min_ratio * price
+    max_rate = max_ratio * price
+
+    ether = (1 / rate) * np.log(price / min_rate)
+    tokens = (1 / rate) * (1 / price - 1 / max_rate)
+    tokens_in_eth = tokens * price
+
+    print("ether: " + str(ether))
+    print("tokens: " + str(tokens))
+    print("tokens_in_eth: " + str(tokens_in_eth))
+
+    return ether, tokens
+
+
+def calc_min_max_ratio_from_balances(
+        price,
+        rate,
+        ether,
+        tokens
+):
+
+    max_rate = price / (1 - rate * price * tokens)
+    min_rate = price / np.exp(rate * ether)
+    min_ratio = min_rate / price
+    max_ratio = max_rate / price
+    print("min_min_ratio: " + str(min_ratio))
+    print("max_max_ratio: " + str(max_ratio) + "\n")
+
+    return min_ratio, max_ratio
+
+
+def calc_rate_from_eth_balance(ether, price, min_ratio):
+    min_rate = min_ratio * price
+    rate = (1 / ether) * np.log(price / min_rate)
+    print("rate: " + str(rate))
+
+
+def calc_params_from_ratios(
+    rate,
+    price,
+    num_formula_precision_bits,
+    max_cap_buy_eth,
+    max_cap_sell_eth,
+    fee_in_precents,
+    min_ratio,
+    max_ratio
+):
+
+    max_rate = max_ratio * price
+    min_rate = min_ratio * price
+
+    _rInFp = rate * (2 ** num_formula_precision_bits)
+    _pMinInFp = min_rate * (2 ** num_formula_precision_bits)
+    _numFpBits = num_formula_precision_bits
+    _maxCapBuyInWei = max_cap_buy_eth * (10 ** 18)
+    _maxCapSellInWei = max_cap_sell_eth * (10 ** 18)
+    _feeInBps = fee_in_precents * 100
+    _maxTokenToEthRateInPrecision = max_rate * (10 ** 18)
+    _minTokenToEthRateInPrecision = min_rate * (10 ** 18)
+
+    print("_rInFp: %.0f" % _rInFp)
```

Comment:
technically, everything here should be in int, not float (%d)

Reply:
yeah you are right, changed to %d.